### PR TITLE
feature/login_enterUserData

### PR DESCRIPTION
**Pull request by ddanilyuk**

### CODE REVIEW

File: Owl/Models/Auth/SetMeSuccess.swift [hunk 1]
```diff
@@ -0,0 +1,13 @@
+//
+//  SetMeSuccess.swift
+//  Owl
+//
+//  Created by Denys Danyliuk on 16.04.2022.
+//
+
+import Foundation
+
+enum SetMeSuccess: Equatable {
```
Comment (AnastasiaHolovash) [inline Owl/Models/Auth/SetMeSuccess.swift]: Change to `SignInUserType`

File: Owl/Models/Auth/UpdateUser.swift [hunk 2]
```diff
@@ -0,0 +1,20 @@
+//
+//  UpdateUser.swift
+//  Owl
+//
+//  Created by Denys Danyliuk on 16.04.2022.
+//
+
+import Foundation
+
+struct UpdateUser {
```
Comment (AnastasiaHolovash) [inline Owl/Models/Auth/UpdateUser.swift]: Rename to `UserUpdate`

File: Owl/Modules/Login/EnterPhone/EnterPhoneView.swift [hunk 3]
```diff
@@ -76,6 +76,7 @@ struct EnterPhoneView: View {
                 dismiss: .dismissAlert
             )
         }
+        .background(Color(UIColor.systemGroupedBackground).edgesIgnoringSafeArea(.all))
```
Comment (AnastasiaHolovash) [inline Owl/Modules/Login/EnterPhone/EnterPhoneView.swift]: Lets use modifiers with new line:
```
.background(
    Color(UIColor.systemGroupedBackground)
        .edgesIgnoringSafeArea(.all)
)
```

File: Owl/Modules/Login/EnterUserData/EnterUserDataStore.swift [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            state.saveButtonEnabled = !state.firstName.isEmpty
+            return .none
+
+        case .later:
+            return .none
+
+        case .save:
+            guard let authUser = environment.authClient.currentUser() else {
+                return .none
+            }
```
Comment (AnastasiaHolovash) [inline Owl/Modules/Login/EnterUserData/EnterUserDataStore.swift:57]: 👍 🫶

File: Owl/Modules/Login/EnterUserData/EnterUserDataView.swift [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+                            .disabled(!viewStore.saveButtonEnabled)
+                        }
+                    }
+                    .padding(20)
+                    .frame(minHeight: proxy.size.height)
+                    .disabled(viewStore.isLoading)
+                    .overlay(
+                        viewStore.isLoading
+                        ? Loader()
+                        : nil
```
Comment (AnastasiaHolovash) [inline Owl/Modules/Login/EnterUserData/EnterUserDataView.swift]: Fix alignment

File: Owl/Modules/Login/EnterUserData/EnterUserDataView.swift [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+                    )
+                    .alert(
+                        self.store.scope(state: \.alert),
+                        dismiss: .dismissAlert
+                    )
+                }
+            }
+        }
+        .background(Color(UIColor.systemGroupedBackground).edgesIgnoringSafeArea(.all))
```
Comment (AnastasiaHolovash) [inline Owl/Modules/Login/EnterUserData/EnterUserDataView.swift]: Lets use modifiers with new line

File: Owl/Modules/Login/LoginStore.swift [hunk 7]
```diff
@@ -66,7 +66,17 @@ struct Login {
...
 
-        case .routeAction(_, action: .enterCode(.setMeResult(.success))):
+        case let .routeAction(_, action: .enterCode(.setMeResult(.success(setMeSuccess)))):
+            switch setMeSuccess {
+            case .newUser:
+                state.routes.push(.enterUserData(.init()))
+                return .none
+            case .userExists:
+                return Effect(value: .delegate(.loginSuccess))
+            }
```
Comment (AnastasiaHolovash) [inline Owl/Modules/Login/LoginStore.swift]: spaces between cases

File: Owl/Modules/Login/Onboarding/OnboardingView.swift [hunk 8]
```diff
@@ -42,6 +42,7 @@ struct OnboardingView: View {
             }
             .padding(20)
         }
+        .background(Color(UIColor.systemGroupedBackground).edgesIgnoringSafeArea(.all))
```
Comment (AnastasiaHolovash) [inline Owl/Modules/Login/Onboarding/OnboardingView.swift]: Lets use modifiers with new line

File: Owl/Modules/Root/Clients/Firestore/FirestoreUsersClient.swift [hunk 9]
```diff
@@ -7,53 +7,64 @@
...
-                            return collection.document()
-                                .setData(from: user, encoder: Firestore.Encoder())
+                        if !snapshot.exists {
+                            return documentRef.setData(from: user)
                                 .eraseToAnyPublisher()
                         } else {
-                            result(.success(true))
+                            result(.success(.userExists))
                             return Empty(completeImmediately: true)
                                 .eraseToAnyPublisher()
```
Comment (AnastasiaHolovash) [inline Owl/Modules/Root/Clients/Firestore/FirestoreUsersClient.swift]: Use `switch snapshot.exists { ... }`

**Review by AnastasiaHolovash — CHANGES_REQUESTED** [review]

File: Owl/Modules/Login/EnterUserData/EnterUserDataView.swift (hunk 6 above)
Comment (ddanilyuk) [reply Owl/Modules/Login/EnterUserData/EnterUserDataView.swift]: Nice

**Review by AnastasiaHolovash — APPROVED** [review]

**Pull request merged**